src/types.ts

```diff
@@ -29,7 +29,7 @@ export interface RequestOptions {
 }
 
 /** @public */
-export type ClientPerspective = 'previewDrafts' | 'published' | 'raw'
+export type ClientPerspective = ('previewDrafts' | 'published' | 'raw') | string
```

Comment:
Is adding string a future-proofing? Cause according to your validator, you dont accept string, so why add the string? Isnt it better to get the compiler to give early error? (keep the validator tho, defensive programming ect)

Reply:
It was to make it less annoying for people to do:
```ts
const config = {
  projectId: 'abc123',
  dataset: 'production',
  perspective: 'published' as const // <- tried to lose the `as const` req here
}

const client = createClient(config)
```
I'll remove it and re-assess later how to handle this papercut